src/LibraryView.tsx

```diff
@@ -17,13 +18,16 @@ export class LibraryView {
     htmlElementId: string = "";
     loadedTypesJson: any = null;
     layoutSpecsJson: any = null;
+    reactor: Reactor = null;
 
     constructor(config: LibraryViewConfig) {
 
         this.setLoadedTypesJson = this.setLoadedTypesJson.bind(this);
         this.setLayoutSpecsJson = this.setLayoutSpecsJson.bind(this);
         this.prefetchContents = this.prefetchContents.bind(this);
         this.updateContentsInternal = this.updateContentsInternal.bind(this);
+        LibraryView.prototype.reactor = new Reactor();
```

Comment:
After reviewing further, I think we need to pass `LibraryView` to all the sub-items (i.e. `LibraryItem` and `ClusterView`) so that those classes can have access to this copy of the `Reactor`. For example, in all the places that we need to raise an event, the following should be done:

```ts
// Now we also need to pass LibraryView to LibraryItem through its property
export interface LibraryItemProps { libraryView: LibraryView, data: ItemData }

export class LibraryItem extends React.Component<LibraryItemProps, LibraryItemState> {

  onSomethingHappened() {
    let libraryView = this.props.libraryView; // Then we obtain the LibraryView like this
    libraryView.raiseEvent("somethingHappened", "some params");
}
```